server/server.go

```diff
@@ -248,6 +256,42 @@ func (server *HTTPServer) voteOnRule(writer http.ResponseWriter, request *http.R
 	}
 }
 
+func (server *HTTPServer) validateClusterAndRuleID(
+	clusterID types.ClusterName, ruleID types.RuleID, writer http.ResponseWriter,
+) bool {
+	clusterExists, err := server.Storage.CheckIfClusterExists(clusterID)
+	if err != nil {
+		log.Err(err)
+		responses.Send(http.StatusInternalServerError, writer, err.Error())
+		return false
+	}
+	if !clusterExists {
+		responses.Send(
+			http.StatusNotFound,
+			writer,
+			(&storage.ItemNotFoundError{ItemID: clusterID}).Error(),
+		)
+		return false
+	}
+
+	ruleExists, err := server.Storage.CheckIfRuleExists(ruleID)
+	if err != nil {
+		log.Err(err)
+		responses.Send(http.StatusInternalServerError, writer, err.Error())
+		return false
+	}
+	if !ruleExists {
+		responses.Send(
```

Comment:
the returned err from responses.Send* is not handled

Reply:
it's gonna be huge refactoring since we're not checking it everywhere :)

Reply:
we are, it was in my PR yesterday. That's also why the `handleServerError` function, so it doesn't have to be everywhere :D

Reply:
+1

Reply:
fixed